tests/runner.py

```diff
@@ -663,9 +664,13 @@ def do_run(self, src, expected_output, args=[], output_nicerizer=None,
     for engine in js_engines:
       #print 'test in', engine
       js_output = self.run_generated_code(engine, filename + '.o.js', args, output_nicerizer=output_nicerizer, assert_returncode=assert_returncode)
+      js_output = js_output.replace('\r\n', '\n')
       try:
-        self.assertContained(expected_output, js_output.replace('\r\n', '\n'))
-        self.assertNotContained('ERROR', js_output)
+        if assert_identical:
+          self.assertIdentical(expected_output, js_output)
+        else:
+          self.assertContained(expected_output, js_output)
+          self.assertNotContained('ERROR', js_output)
```

Comment:
If the last test in test_getdents64_special_cases fails because of a partial character being written at the end of the string (e.g. if stringToUTF8 is broken), its output will contain the expected output plus a garbage character and assertContained will succeed. For this not to happen I could:
1) Reorder the calls to doTest. This is too non-obvious and unreliable IMO,
2) Print something at the end of the test (e.g. "done") and add the same string into the expected output. This is how I did it initially, but it seemed rather unreliable too - the essential part here is the "done" string in the expected output and I can't put a comment there to mention its importance.
So, introducing the "assert_identical" mode looked the best to me.